ee/vellum_ee/workflows/display/utils/expressions.py

```diff
@@ -275,16 +275,25 @@ def serialize_value(executable_id: UUID, display_context: "WorkflowDisplayContex
         return serialize_value(executable_id, display_context, child_descriptor)
 
     if isinstance(value, WorkflowInputReference):
-        try:
-            workflow_input_display = display_context.global_workflow_input_displays[value]
-        except KeyError:
-            inputs_class_name = value.inputs_class.__name__
-            workflow_class_name = display_context.workflow_display_class.infer_workflow_class().__name__
-            raise UnsupportedSerializationException(
-                f"Inputs class '{inputs_class_name}' referenced during serialization of '{workflow_class_name}' "
-                f"without parameterizing this Workflow with this Inputs definition. Update your Workflow "
-                f"definition to '{workflow_class_name}(BaseWorkflow[{inputs_class_name}, BaseState])'."
-            )
+        if value not in display_context.global_workflow_input_displays:
+            if not value.types:
+                raise InvalidInputReferenceError(
+                    message=f"type object '{value.inputs_class.__qualname__}' has no attribute '{value.name}'",
+                    inputs_class_name=value.inputs_class.__qualname__,
+                    attribute_name=value.name,
+                )
+            else:
+                workflow_inputs_class = display_context.workflow_display_class.infer_workflow_class().get_inputs_class()
+                if value.inputs_class != workflow_inputs_class:
```

Comment:
add an else case here, raising the same `InvalidInputReferenceError` as above. In fact, idt we need the `if not value.types:` if/else, just this one.